### PR TITLE
chore(codecs): consolidate enum types

### PR DESCRIPTION
**Pull request by dsmith3197**
Consolidate enum types to reduce duplicate type definitions and documentation.

### CODE REVIEW

File: lib/codecs/src/decoding/format/mod.rs [hunk 1]
```diff
@@ -44,3 +46,8 @@ dyn_clone::clone_trait_object!(Deserializer);
 
 /// A `Box` containing a `Deserializer`.
 pub type BoxedDeserializer = Box<dyn Deserializer>;
+
+/// Default value for the UTF-8 lossy option.
+const fn default_lossy() -> bool {
+    true
+}
```
Comment (spencergilbert) [inline lib/codecs/src/decoding/format/mod.rs:53]: I know we have a shared function for [`default_true`](https://github.com/vectordotdev/vector/blob/c97d619d47b1171d592dcf55692b5caa01e97992/src/serde.rs#L11) - it's specifically for `serde`, but maybe could be reused?

**Review by spencergilbert — COMMENTED** [review]
Small note on the `default_lossy` function, and some amusement that the consolidation has such a positive diff of new lines in the PR 😆

**Review by spencergilbert — COMMENTED** [review]
Small note on the `default_lossy` function, and some amusement that the consolidation has such a positive diff of new lines in the PR 😆

**Comment by spencergilbert** [comment]
> I needed to rebase 😆

Ah, duh - I thought this had been targeting your other branch. Makes more sense now!

**Review by spencergilbert — APPROVED** [review]
Much more consolidation than before rebasing 😂

File: lib/codecs/src/decoding/format/mod.rs (hunk 1 above)
Comment (dsmith3197) [reply lib/codecs/src/decoding/format/mod.rs]: Yeah, I considered that. The issue is that `default_true()` is defined in `vector`, meaning that we can't import it in `codecs` without a circular dependency. We could move it to `vector_corde::serde` but it didn't seem worth the effort at the time. What do you think?

Comment (spencergilbert) [reply lib/codecs/src/decoding/format/mod.rs]: Eh, probably not worth it - I remember wishing before you could more easily `default = bool::True` without so much hassle.

**Pull request merged**